### ISSUE TITLE
`KeycloakServiceAccountAdminTokenRetriever` vs `KeycloakAdminToken`

### ISSUE THREAD

**Issue by omid** [kilork/keycloak#111]
Better to follow the same pattern for both structs.

One has `acquire` the other one has `create`.
One the `get` sends a request always, in the other one uses the `acquired` version.

Basically, we added `grant_type` to `KeycloakAdminToken` to support service accounts, and now we have `KeycloakServiceAccountAdminTokenRetriever` which does the same, kinda.

**Comment by ovnicraft** [comment]
I'm working on getting admin token for service account but I found `acquire` is private, we need it public

https://github.com/kilork/keycloak/blob/master/src/rest/mod.rs#L66

**Comment by omid** [comment]
@ovnicraft I copy pasted the code and have my own logic! 🤷🏻

**Comment by ovnicraft** [comment]
> @ovnicraft I copy pasted the code and have my own logic! 🤷🏻

Yes, btw that struct has `create()` and `acquire`, i would like to use it.
